platform/iphone/app_delegate.mm

```diff
@@ -720,15 +187,50 @@ - (void)applicationWillTerminate:(UIApplication *)application {
 // notification panel by swiping from the upper part of the screen.
 
 - (void)applicationWillResignActive:(UIApplication *)application {
-	on_focus_out(view_controller, &is_focus_out);
+	[self toggleFocus:NO];
 }
 
 - (void)applicationDidBecomeActive:(UIApplication *)application {
-	on_focus_in(view_controller, &is_focus_out);
+	[self toggleFocus:YES];
+}
+
+- (void)toggleFocus:(BOOL)isFocused {
+	if (_focused != isFocused) {
+		_focused = isFocused;
+
+		// Handle notifications to main loop
+		if (OS::get_singleton()->get_main_loop()) {
+			int notification = self.isFocused ? MainLoop::NOTIFICATION_WM_FOCUS_IN : MainLoop::NOTIFICATION_WM_FOCUS_OUT;
+			OS::get_singleton()->get_main_loop()->notification(notification);
+		}
+
+		// OpenGL Animation
+		GodotGameViewController *viewController = (GodotGameViewController *)self.window.rootViewController;
+		if (self.isFocused) {
+			[viewController.glView startAnimation];
+		} else {
+			[viewController.glView stopAnimation];
+		}
+
+		// Native Video
+		if (OS::get_singleton()->native_video_is_playing()) {
```

Comment:
Not sure if this is still much in use, everything I found was quite deprecated, but we do have some fancy new APIs that could replace them.